src/Domain/CoordinateMaps/BulgedCube.hpp

```diff
@@ -0,0 +1,469 @@
+ *
+ *
+ *  ### The Special Case for Octahedral Symmetry
+ *  The general formula is for the case in which there are six independently
+ *  specified bounding surfaces. In our case, the surfaces are obtained by
+ *  rotations and reflections of the upper-\f$\zeta\f$ face.
+ *
+ * We define the matrices corresponding to these transformations to be:
+ *
+ * \f[S_{xy} =
+ *  \begin{bmatrix}
+ *  0 & 1 & 0\\
+ *  1 & 0 & 0\\
+ *  0 & 0 & 1\\
+ *  \end{bmatrix},\
+ *
+ * S_{xz} =
+ *  \begin{bmatrix}
+ *  0 & 0 & 1\\
+ *  0 & 1 & 0\\
+ *  1 & 0 & 0\\
+ *  \end{bmatrix},\
+ *
+ * S_{yz} =
+ *  \begin{bmatrix}
+ *  1 & 0 & 0\\
+ *  0 & 0 & 1\\
+ *  0 & 1 & 0\\
+ *  \end{bmatrix}\f]
+ *
+ * \f[C_{zxy} =
+ *  \begin{bmatrix}
+ *  0 & 0 & 1\\
+ *  1 & 0 & 0\\
+ *  0 & 1 & 0\\
+ *  \end{bmatrix},\
+ *
+ * C_{yzx} =
+ *  \begin{bmatrix}
+ *  0 & 1 & 0\\
+ *  0 & 0 & 1\\
+ *  1 & 0 & 0\\
+ *  \end{bmatrix}\f]
+ *
+ * \f[N_{x} =
+ *  \begin{bmatrix}
+ *  -1 & 0 & 0\\
+ *  0 & 1 & 0\\
+ *  0 & 0 & 1\\
+ *  \end{bmatrix},\
+ *
+ * N_{y} =
+ *  \begin{bmatrix}
+ *  1 & 0 & 0\\
+ *  0 & -1 & 0\\
+ *  0 & 0 & 1\\
+ *  \end{bmatrix},\
+ *
+ * N_{z} =
+ *  \begin{bmatrix}
+ *  1 & 0 & 0\\
+ *  0 & 1 & 0\\
+ *  0 & 0 & -1\\
+ *  \end{bmatrix}\f]
+ *
+ * The surface maps can now all be written in terms of
+ * \f$\vec{\sigma}_{+\zeta}\f$ and these matrices:
+ * <center>
+ * \f$\vec{\sigma}_{-\zeta}(\xi, \eta) = N_z\vec{\sigma}_{+\zeta}(\xi, \eta)\\
+ * \vec{\sigma}_{+\eta}(\xi, \zeta) = S_{yz}\vec{\sigma}_{+\zeta}(\xi, \zeta)\\
+ * \vec{\sigma}_{-\eta}(\xi, \zeta) = N_yS_{yz}\vec{\sigma}_{+\zeta}(\xi,
+ * \zeta)\\
+ * \vec{\sigma}_{+\xi}(\eta, \zeta) = C_{zxy}\vec{\sigma}_{+\zeta}(\eta,
+ * \zeta)\\
+ * \vec{\sigma}_{-\xi}(\eta, \zeta) = N_xC_{zyx}\vec{\sigma}_{+\zeta}(\eta,
+ * \zeta)\f$
+ * </center>
+ *
+ * The four bounding curves \f$\vec{\Gamma}\f$ on the \f$+\zeta\f$ face are
+ * given by:
+ *
+ * <center>
+ * \f$\vec{\Gamma}_{+\xi,+\zeta}(\eta) = \vec{\sigma}_{+\zeta}(+1,\eta)\\
+ * \vec{\Gamma}_{-\xi,+\zeta}(\eta) = \vec{\sigma}_{+\zeta}(-1,\eta)
+ * = N_x\vec{\sigma}_{+\zeta}(+1, \eta)\\
+ * \vec{\Gamma}_{+\eta,+\zeta}(\xi) = \vec{\sigma}_{+\zeta}(\xi,+1)
+ * = S_{xy}\vec{\sigma}_{+\zeta}(+1, \xi)\\
+ * \vec{\Gamma}_{-\eta,+\zeta}(\xi) = \vec{\sigma}_{+\zeta}(\xi,-1)
+ * = N_yS_{xy}\vec{\sigma}_{+\zeta}(+1,\xi)\f$
+ * </center>
+ *
+ * The bounding curves on the other surfaces can be obtained by transformations
+ * on the \f$+\zeta\f$ face:
+ *
+ * <center>
+ * \f$\vec{\Gamma}_{+\xi,-\zeta}(\eta) = N_z\vec{\sigma}_{+\zeta}(+1,\eta)\\
+ * \vec{\Gamma}_{-\xi,-\zeta}(\eta) = N_z\vec{\sigma}_{+\zeta}(-1,\eta)
+ * = N_zN_x\vec{\sigma}_{+\zeta}(+1,\eta)\\
+ * \vec{\Gamma}_{+\eta,-\zeta}(\xi) = N_z\vec{\sigma}_{+\zeta}(\xi,+1)
+ * = N_zS_{xy}\vec{\sigma}_{+\zeta}(+1, \xi)\\
+ * \vec{\Gamma}_{-\eta,-\zeta}(\xi) = N_z\vec{\sigma}_{+\zeta}(\xi,-1)
+ * = N_zN_yS_{xy}\vec{\sigma}_{+\zeta}(+1, \xi)\f$
+ * </center>
+ *
+ * <center>
+ * \f$\vec{\Gamma}_{+\xi,+\eta}(\zeta) =
```

Comment:
This seems insanely long for inline math...

Reply:
Perhaps some of this math can be moved from the C++ documentation into a companion latex file?

Reply:
I feel very, very, very strongly against this. That's done in SpEC and it's largely unmaintainable because nobody knows where there are latex files, our collaboration seems to be terrible at using standard packages, and if we want to include them into the documentation we need to find them all, build them all (non-trivial), and then link them all. I realize we might need this at some point, but anything that we can keep in the C++ code itself is good. It's most likely to get updated and most likely to actually be read.

Reply:
Enforcing a restriction to standard packages is relatively easy: have Travis compile the TeX files.